ecal_core uses C++20 / C++17 std::transform

### Problem Description

eCAL Master currently cannot be compiled with C++14 any more. The error is:

```
/home/florian/Projects/ecal-launchpad/test_build/ecal-100.0.0/ecal/core/src/pubsub/ecal_subgate.cpp: In member function ‘bool eCAL::CSubGate::ApplySample(const eCAL::pb::Sample&, eCAL::pb::eTLayerType)’:
/home/florian/Projects/ecal-launchpad/test_build/ecal-100.0.0/ecal/core/src/pubsub/ecal_subgate.cpp:166:14: error: ‘transform’ is not a member of ‘std’
         std::transform(
              ^~~~~~~~~
/home/florian/Projects/ecal-launchpad/test_build/ecal-100.0.0/ecal/core/src/pubsub/ecal_subgate.cpp: In member function ‘bool eCAL::CSubGate::ApplySample(const string&, const string&, const char*, size_t, long long int, long long int, long long int, size_t, eCAL::pb::eTLayerType)’:
/home/florian/Projects/ecal-launchpad/test_build/ecal-100.0.0/ecal/core/src/pubsub/ecal_subgate.cpp:210:12: error: ‘transform’ is not a member of ‘std’
       std::transform(
            ^~~~~~~~~
ecal/core/CMakeFiles/core.dir/build.make:456: recipe for target 'ecal/core/CMakeFiles/core.dir/src/pubsub/ecal_subgate.cpp.o' failed
make[2]: *** [ecal/core/CMakeFiles/core.dir/src/pubsub/ecal_subgate.cpp.o] Error 1
```

I am not sure, whether we are using a C++17 or C++20 feature here. Some overloads have been added with C++17, some with C++20:
https://en.cppreference.com/w/cpp/algorithm/transform

### How to reproduce

Compile eCAL on Ubuntu 18.04. Ubuntu 18.04 should support C++17 but not 20, which is why I assume that we currently depend on a C++20 compiler.

### How did you get eCAL?

Custom Build / Built from source

### Environment

eCAL Master
Ubuntu 18.04
gcc (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0

### eCAL System Information

_No response_

--- a/ecal/core/src/mon/ecal_monitoring_impl.cpp
+++ b/ecal/core/src/mon/ecal_monitoring_impl.cpp
@@ -94,7 +94,7 @@ namespace eCAL
     m_host_name = Process::GetHostName();
 
     // utilize registration receiver to enrich monitor information
-    g_registration_receiver()->SetCustomApplySampleCallback([this](const auto& ecal_sample_){ApplySample(ecal_sample_, eCAL::pb::tl_none);});
+    g_registration_receiver()->SetCustomApplySampleCallback([this](const auto& ecal_sample_){this->ApplySample(ecal_sample_, eCAL::pb::tl_none);});
 
     // start logging receive thread
     const CLoggingReceiveThread::LogMessageCallbackT logmsg_cb = std::bind(&CMonitoringImpl::RegisterLogMessage, this, std::placeholders::_1);

--- a/ecal/core/src/pubsub/ecal_subgate.cpp
+++ b/ecal/core/src/pubsub/ecal_subgate.cpp
@@ -21,6 +21,8 @@
  * @brief  eCAL subscriber gateway class
 **/
 
+#include <algorithm>
+
 #include <ecal/ecal.h>
 
 #ifdef ECAL_OS_LINUX

--- a/ecal/core/src/readwrite/ecal_reader.cpp
+++ b/ecal/core/src/readwrite/ecal_reader.cpp
@@ -38,6 +38,7 @@
 #include <sstream>
 #include <iostream>
 #include <utility>
+#include <algorithm>
 
 namespace eCAL
 {

